### DISCUSSION TITLE
Pemilihan Variable Predictor Menggunakan Nilai AIC

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#737] [Q&A] [answered]
Dear rekan2 Algoritma,

Mohon saran dan petunjuknya, saya sedang membuat pemodelan untuk logistik resgression, dimana untuk menentukan variable predictor yang akan digunakan saya menggunakan metode nilai AIC dari setiap variabel yang ada. Hasil yang diperoleh sebagai berikut:

<img width="261" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/b0353d24-9460-4a65-ba34-96ae03d7d6ec">

Jika dilihat dari hasilnya rentang nilai AIC antara 4 sampai dengan 38, bagaimana menentukan variabel-variabel apa yang akan dipilih jika melihat dari nilainya. Kemudian apakah metode yang saya gunakan dalam pemilihan variabel sudah cukup atau ada cara lain yang lebih baik.

**Comment by divakartika** [comment]
Dear @joejnt 

Jika menggunakan stepwise regression, maka algoritma ini akan melihat kenaikan & penurunan nilai AIC dari penambahan / pengurangan prediktor
- Forward step: dari model tanpa prediktor, ditambahkan prediktor satu-persatu, dilihat apakah penambahan prediktor meningkatkan / menurunkan AIC. Jika meningkatkan AIC maka proses akan berhenti

![image](https://github.com/teamalgoritma/community/assets/79572421/38d22a65-a550-4967-ba43-45213f75760e)

- Backward step: dari model dengan seluruh prediktor, prediktor dikurangi satu-persatu. Jika pengurangan prediktor meningkatkan AIC maka proses akan berhenti

![image](https://github.com/teamalgoritma/community/assets/79572421/dd50633b-0b72-4641-9156-2a6ff008e9ef)

Kombinasi prediktor yang dihasilkan dari proses stepwise merupakan kombinasi yang menghasilkan AIC terendah. Kombinasi ini dapat dilihat dari bagian Final Model. Dari screenshot yang Anda berikan, model terbaik hasil stepwise adalah `gb_flag ~ dpd`, artinya model dengan AIC terkecil hanya menggunakan prediktor `dpd`.

Cara ke-dua untuk memilih variabel adalah dengan melihat signifikansi. Jika p-value prediktor < 0.05, maka prediktor ini signifikan. P-value prediktor bisa didapatkan dari hasil `summary(model)`. Prediktor dengan p-value < 0.05 memiliki tanda `*` di sebelahnya.

![image](https://github.com/teamalgoritma/community/assets/79572421/101096e8-7994-4b62-9c99-0f9e78d417a3)

Cara ke-tiga adalah dengan business knowledge. Jika secara bisnis suatu prediktor dirasa penting, maka prediktor tersebut perlu diikutsertakan dalam pembuatan model.

Semoga membantu dan menjawab pertanyaan Anda :)

**Reply by joejnt** [reply]
Dear @divakartika 

Terima kasih untuk sarannya, setelah saya cek summary dari model hasilnya seperti berikut:
<img width="249" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/9af0b2e9-43c0-48fe-96f2-161328203ae8">


Dilihat dari nilai p-value nya, hanya ada 1 variable yang nilainya NA, sedangkan var yang lainnya memiliki nilai p-value > 0,05. Jika ditemukan kondisi seperti ini, bagaimana langkah terbaik dalam menentukan var predictor nya.

Sementara ini, saya melakukan langkah dengan menghitung nilai IV 
syntax:
iv_smt <- map_df(fineclass_smt, ~pluck(.x, 10, 1)) %>% 
  pivot_longer(everything(), names_to = "variable", values_to = "iv")

kemudian membuat cluster, hasilnya seperti berikut:
<img width="294" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/4a9becac-6ac3-4f05-ab12-e3865503136c">

cara saya adalah dengan memilih masing-masing 1 atau 2 variabel dari setiap cluster yang ada tujuannya untuk mengurangi noise.

Namun kendalanya adalah bagaimana menentukan variable yang tepat pada setiap clusternya.

Best regards
JT

**Reply by divakartika** [reply, marked as answer]
Dear @joejnt  ,

Sebelumnya mohon maaf atas kekeliruan pada jawaban saya sebelumnya, variabel prediktor yang signifikan memiliki p-value < 0.05, sehingga terdapat tanda * di sebelahnya.

Dari screenshot summary model yang dilampirkan, terlihat bahwa tidak ada variabel yang signifikan. Residual Deviance juga bernilai sangat jauh dari Null Deviance. Hal ini dapat terjadi akibat perfect separation. Dapat dicoba untuk mengeluarkan salah satu / lebih prediktor secara bergantian dan dilihat kembali summary model yang dihasilkan. Kemungkinan prediktor yang menyebabkan perfect separation adalah prediktor dengan Estimate yang sangat kecil atau sangat besar.

Semoga membantu dan menjawab pertanyaan Anda.

**Comment by joejnt** [comment]
Dear @divakartika 

Terima kasih untuk insidenya

Best regards,
JT
